### PR TITLE
EES-1346 Add tests to check latest release is correct

### PR DESCRIPTION
**Pull request by markhiveit**

### CODE REVIEW

**Comment by markhiveit** [comment]
Nick suggested not creating a new test suite and instead performing this test at the end of a different test suite, so I'll make that change when I get time.

File: tests/robot-tests/tests/admin_and_public/bau/publish_data.robot [hunk 1]
```diff
@@ -161,6 +202,31 @@ Verify newly published release is on Find Statistics page
...
     user checks publication bullet does not contain link  ${PUBLICATION_NAME}   Statistics at DfE
 
+Navigate to published release page
+    [Tags]  HappyPath
+    user clicks testid element   View stats link for ${PUBLICATION_NAME}
+    user waits until h1 is visible   ${PUBLICATION_NAME}  90
+
+Check latest release is correct
+    [Tags]  HappyPath
+    user checks testid element contains  page-title-caption    Financial Year 3001-02
```
Comment (ntsim) [inline tests/robot-tests/tests/admin_and_public/bau/publish_data.robot:212]: We have a keyword for this: `user waits until page contains title caption`

File: tests/robot-tests/tests/admin_and_public/bau/publish_data.robot [hunk 2]
```diff
@@ -161,6 +202,31 @@ Verify newly published release is on Find Statistics page
...
+
+    user opens details dropdown  See 1 other releases
+    user checks page contains other release   Financial Year 3000-01
+    user checks page does not contain other release   Financial Year 3001-02
+
+    user clicks link   Financial Year 3000-01
+
+Check other release is correct
+    [Tags]  HappyPath
+    user checks testid element contains   page-title-caption   Financial Year 3000-01
```
Comment (ntsim) [inline tests/robot-tests/tests/admin_and_public/bau/publish_data.robot:224]: We have a keyword for this: `user waits until page contains title caption`

File: tests/robot-tests/tests/libs/utilities.py [hunk 3]
```diff
@@ -343,3 +343,7 @@ def remove_substring_from_right_of_string(string, substring):
         return string[:-len(substring)]
     else:
         raise_assertion_error(f'String "{string}" doesn\'t end with substring "{substring}"')
+
+def user_clicks_element_if_exists(selector):
```
Comment (ntsim) [inline tests/robot-tests/tests/libs/utilities.py:347]: I think we could just do this with Robot keywords fairly easily

File: tests/robot-tests/tests/admin_and_public/bau/publish_data.robot [hunk 4]
```diff
@@ -19,24 +19,65 @@ Create new publication for "UI tests topic" topic
...
+    user clicks link   Release status
+    user waits until h2 is visible  Release status
+    user waits until page contains button  Edit release status
+
+Approve release
+    [Tags]  HappyPath
+    user clicks button  Edit release status
+    user waits until h2 is visible  Edit release status
+
+    user clicks element   css:input[data-testid="Approved for publication"]
```
Comment (ntsim) [inline tests/robot-tests/tests/admin_and_public/bau/publish_data.robot:44]: `user clicks radio` would be better

File: tests/robot-tests/tests/admin_and_public/bau/publish_data.robot [hunk 5]
```diff
@@ -19,24 +19,65 @@ Create new publication for "UI tests topic" topic
...
+    user enters text into element  id:releaseStatusForm-internalReleaseNote  Approved by UI tests - latest release
+    user clicks element  css:input[data-testid="As soon as possible"]
```
Comment (ntsim) [inline tests/robot-tests/tests/admin_and_public/bau/publish_data.robot:46]: `user clicks radio` would be better

**Review by ntsim — CHANGES_REQUESTED** [review]

File: tests/robot-tests/tests/libs/utilities.py (hunk 3 above)
Comment (markhiveit) [reply tests/robot-tests/tests/libs/utilities.py:347]: Discussed on Slack. Tried using `run keyword and ignore error`, but a screenshot is still getting created when the keyword fails, so going to stick with the python function.

**Review by ntsim — APPROVED** [review]
Approved although it looks like the build is failing?

**Pull request merged**